#### Problem Description - To minimize the Total variance(risk) in investment

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
plt.style.use('ggplot')
import seaborn as sns
from sklearn import *

/Users/dkhandelwal/anaconda/lib/python2.7/site-packages/sklearn/lda.py:4: DeprecationWarning: lda.LDA has been moved to discriminant_analysis.LinearDiscriminantAnalysis in 0.17 and will be removed in 0.19
  "in 0.17 and will be removed in 0.19", DeprecationWarning)
/Users/dkhandelwal/anaconda/lib/python2.7/site-packages/sklearn/qda.py:4: DeprecationWarning: qda.QDA has been moved to discriminant_analysis.QuadraticDiscriminantAnalysis in 0.17 and will be removed in 0.19.
  "in 0.17 and will be removed in 0.19.", DeprecationWarning)


In [3]:
# Loading the Data

df = pd.read_csv("Portfolio.csv")

In [4]:
df.head()

X       Y
0  115.14  107.22
1  113.03  109.02
2  111.74  103.08
3  110.31  101.99
4  108.06   99.57

#### Data Set description
- investment in 2 Asset classes X and Y
- we want to compute the accuracy of statistic ,alpha = Y.var())-covar/(df.X.var())+(df.Y.var())-2*covar


In [38]:
# Defining function that calculates statistics of interest
def alpha_func(df):
    df_bootstrapped = df.sample(frac = .8 , replace = True)
    X_var = df_bootstrapped.X - df_bootstrapped.X.mean()
    Y_var = df_bootstrapped.Y - df_bootstrapped.Y.mean()
    covar = ((X_var*Y_var).sum())/len(df)   # Calculaitng the covariance
    
    return((df.Y.var())-covar/(df.X.var())+(df.Y.var())-2*covar)

In [39]:
#running a for loop that would run multiple time to compute the bootstrapped estimates

alpha = []  # Empty list to hold the values
for i in range(1,1000):
    alpha.append(alpha_func(df))

In [42]:
np.std(alpha)

73.678062831211278

###  Conclusion- 
- The standard deviation for this Statistic of interest = 73.68
- This can be interprted as follows:
    - On an avg the estimated value of alpha(alpha_hat) would be off from true alpha by 73.68 units 

